src/codegen/mod.rs

```diff
@@ -987,37 +987,39 @@ impl CodeGenerator for TemplateInstantiation {
         // just converted to rust types in fields, vars, etc, we take this
         // opportunity to generate tests for their layout here.
 
-        let layout = item.kind().expect_type().layout(ctx);
+        if ctx.options().layout_tests {
```

Comment:
Rather than indenting tons of code, lets just early return here if we *aren't* generating layout tests.